File: Divide Two Integers.py
#https://www.naukri.com/code360/problems/divide-two-integers_1112617

from typing import List

def divideTwoInteger(dividend: int, divisor: int) -> int:
    # write your code here
    manipulate=0
    if dividend<0 and divisor>0:
        manipulate=1
        dividend=abs(dividend)
    elif dividend>0 and divisor<0:
        manipulate=1
        divisor=abs(divisor)
    elif divisor<0 and dividend<0:
        divisor=abs(divisor)
        dividend=abs(dividend)
    result=0
    value=0
    while value<dividend:
        if dividend-value<divisor:
            break
        result+=1
        value+=divisor
    if manipulate==1:
        return 0-result
    return result

File: Divide Two Integers.cpp
//https://leetcode.com/problems/divide-two-integers/description/


class Solution {
public:
    int divide(int n, int d) {
        if(d==1) return n;
        long long count=0;
        int sign=0;
        long dividend=(long)n,divisor=(long)d;
        if(n<0 && d<0){
            sign=0;
        }
        else if(n<0 || d<0){
            sign=1;
        }
        dividend=abs((long)n);
        divisor=abs((long)d);
        

        while (dividend>=divisor){
            for(int i=0;i<32;i++){
                if (dividend<=(divisor<<i)){
                    if(i==0){
                        count+=1;
                        dividend=dividend-divisor;
                    }
                    else{
                        count+=1<<(i-1);
                        if(count>INT_MAX) return INT_MAX;
                        else if(count<INT_MIN) return INT_MIN; 
                        dividend=dividend-(divisor<<(i-1));
                    }
                    break;
                }
            }
        }
        if(count>INT_MAX) return INT_MAX;
        else if(count<INT_MIN) return INT_MIN;
        if(sign==1) return -count;
        return count;
    }
};
